safe/gisv4/vector/aggregate_summary.py

```diff
@@ -0,0 +1,132 @@
+# coding=utf-8
+
+"""
+Aggregate the impact table to the aggregate hazard.
+"""
+
+from qgis.core import QGis, QgsField, QgsFeatureRequest
+
+from safe.common.exceptions import InvalidKeywordsForProcessingAlgorithm
+from safe.definitionsv4.fields import (
+    aggregation_id_field,
+    hazard_id_field,
+    exposure_class_field,
+    total_field,
+    exposure_count_field,
+    size_field
+)
+from safe.utilities.profiling import profile
+from safe.utilities.pivot_table import FlatTable
+
+__copyright__ = "Copyright 2016, The InaSAFE Project"
+__license__ = "GPL version 3"
+__email__ = "info@inasafe.org"
+__revision__ = '$Format:%H$'
+
+
+@profile
+def aggregate_summary(aggregate_hazard, impact, callback=None):
+    """Compute the summary from the impact into the aggregate hazard.
+
+    :param aggregate_hazard: The aggregate hazard vector layer.
+    :type aggregate_hazard: QgsVectorLayer
+
+    :param impact: The impact vector layer.
+    :type impact: QgsVectorLayer
+
+    :param callback: A function to all to indicate progress. The function
+        should accept params 'current' (int), 'maximum' (int) and 'step' (str).
+        Defaults to None.
+    :type callback: function
+
+    :return: The new aggregate hazard with summary.
+    :rtype: QgsVectorLayer
+
+    .. versionadded:: 4.0
+    """
+    impact_keywords = impact.keywords
```

Comment:
Would it be simpler if we just refer to `impact.keywords` directly? I saw there is only one reading for `impact_keywords` in line 48.